test/type/explicit_existential.swift

```diff
@@ -202,3 +202,15 @@ func typealiasMemberReferences(metatype: Wrapper.Type) {
   let _: Wrapper.E.Protocol = metatype.E.self
   let _: (any Wrapper.E).Type = metatype.E.self
 }
+
+func testAnyTypeExpr() {
+  let _: (any P).Type = (any P).self
+
+  func test(_: (any P).Type) {}
+  test((any P).self)
```

Comment:
It might be worth testing a few other cases like

    (any P) & (any Q)
    any (P & Q)

and some invalid ones like

    any foo()

etc